### ISSUE TITLE
How could rewrite_as_sin be implemented?

### ISSUE THREAD

**Issue by rikardn** [symengine/symengine#1774]
I was having a look into `rewrite` and I have a question around how `rewrite_as_sin` could be done.

In sympy:

```python
>>> import sympy

>>> x = sympy.Symbol('x')
>>> expr = sympy.cos(x)
>>> expr.rewrite(sympy.sin)
sin(x + pi/2)
```

If this expression would be created in symengine it would be auto-simplified back to cos(x). Since symengine assume all expressions to be in canonical form, I don't see how this non-canonical expression can be created. Any ideas on how to support this kind of rewrite in symengine?

**Comment by isuruf** [comment]
You'll have to create `sin(UnevaluatedExpr(x+pi/2))`

**Comment by rikardn** [comment]
Thanks @isuruf! This works. I have implemented `rewrite_as_sin` in #1778. Discussion can be continued there if needed.
